src/Modules/UngradedModules.jl

```diff
@@ -3877,17 +3878,206 @@ function free_resolution(F::FreeMod)
   return res
 end
 
+#= Fill functions (and helpers) for Hecke ChainComplexes in terms of free resolutions =#
+function get_last_map_key(cc::Hecke.ChainComplex)
+  ctr = cc.start
+  while haskey(cc.maps, ctr)
+    ctr -= 1
+  end
+  return ctr+1
+end
+
+function extend_free_resolution(cc::Hecke.ChainComplex, idx::Int; algorithm::String="fres")
+  key = get_last_map_key(cc)
+  if cc.complete == true
+    return cc.maps[key]
+  end
+
+  kernel_entry          = image(cc.maps[key])[1]
+  br                    = base_ring(kernel_entry)
+  singular_free_module  = singular_module(ambient_free_module(kernel_entry))
+  singular_kernel_entry = Singular.Module(base_ring(singular_free_module),
+                              [singular_free_module(repres(g)) for g in gens(kernel_entry)]...)
+  singular_kernel_entry.isGB = true
+
+  len = idx + key - cc.start + 1
+  if algorithm == "fres"
+    res = Singular.fres(singular_kernel_entry, len, "complete")
+  elseif algorithm == "sres"
+    res = Singular.fres(singular_kernel_entry, len)
+  elseif algorithm == "lres"
+    error("LaScala's method is not yet available in Oscar.")
+  end
+
+  if Singular.length(res) < len
+    cc.complete = true
+  end
+  dom = domain(cc.maps[key])
+  j   = 2
+  while j <= Singular.length(res)
+    key -= 1
+    codom = dom
+    dom   = free_module(br, Singular.ngens(res[j]))
+    SM    = SubModuleOfFreeModule(codom, res[j])
+    generator_matrix(SM)
+    map = hom(dom, codom, SM.matrix)
+    cc.maps[key] = map
+    j += 1
+  end
+  # Finalize maps.
+  if cc.complete == true
+    key -= 1
+    Z = FreeMod(br, 0)
+    set_attribute!(Z, :name => "Zero")
+    cc.maps[key] = hom(Z, domain(cc.maps[key+1]), FreeModElem[])
+  end
+  return cc.maps[key]
+end
+
+@doc Markdown.doc"""
+    free_resolution(M::SubQuo; ordering::ModuleOrdering = default_ordering(M),
+        length::Int=0, method::String="complete", algorithm::String="fres")
+
+Return a free resolution of `M`.
+
+If `length != 0`, the free resolution is only computed up to the `length`-th free module.
+`algorithm` can be set to `sres` or `fres`.
```

Comment:
This does not explain what `method` is?

Reply:
That was a copy and paste error, there is no `method` anymore. Fixed this.